File: slot-dart/lib/src/views/reels_view.dart
library slot.views.reels_view;

import 'package:slot/src/mvc/view.dart';
import 'package:slot/src/models/reels_model.dart';
import 'package:slot/src/models/reel_model.dart';
import 'package:slot/src/views/reel_view.dart';

class ReelsView extends View {

  List<ReelView> _reelViews;

  ReelsView(Model model) : super(model) {

    _reelViews = new List<ReelViews>();

    _createReelViews();

  }

  void _createReelViews() {

    List<ReelModel> reelModels = model.reelModels;

    for (ReelModel reelModel in reelModels) {

      ReelView reelView = new ReelView(reelModel);
      reelView.x = reelModels.indexOf(reelModel) * reelView.width;
      addChild(reelView);
      _reelViews.add(reelView);

    }

  }

  List<ReelView> get reelViews {
    return _reelViews;
  }

  ReelsModel get model {
    return super.model as ReelsModel;
  }

}

File: slot-ts/src/views/ReelsView.ts
import Model from '../engine/mvc/Model';
import View from '../engine/mvc/View';
import ReelModel from '../models/ReelModel';
import ReelsModel from '../models/ReelsModel';
import ReelView from './ReelView';

export default class ReelsView extends View {
  private reelViews: ReelView[];

  constructor(model: Model) {
    super(model);
    this.reelViews = [];
    this.createReelViews();
  }

  private createReelViews(): void {
    const reelModels: ReelModel[] = this.getModel().getReelModels();

    for (const reelModel of reelModels) {
      const reelView: ReelView = new ReelView(reelModel);
      reelView.scaleX = reelView.scaleY = 6;
      reelView.x = reelModels.indexOf(reelModel) * reelView.width + 40;
      reelView.y = -130;
      this.addChild(reelView);

      this.reelViews.push(reelView);
    }
  }

  public getReelViews(): ReelView[] {
    return this.reelViews;
  }

  public getModel(): ReelsModel {
    return this.model as ReelsModel;
  }
}
